### PR TITLE
port mkl binclass example

### PR DESCRIPTION
**Pull request by LiuYuHui**

### CODE REVIEW

File: src/shogun/features/CombinedFeatures.cpp [hunk 1]
```diff
@@ -162,6 +162,11 @@ void CombinedFeatures::init()
 {
 	SG_ADD(&num_vec, "num_vec", "Number of vectors.");
 	SG_ADD(&feature_array, "feature_array", "Feature array.");
+	add_callback_function(
+		"feature_array", [&]() { 
+			if (feature_array.size() > 0) 
+				num_vec = feature_array[0]->get_num_vectors(); 
+	});
 }
```
Comment (LiuYuHui) [inline src/shogun/features/CombinedFeatures.cpp]: when a feature is added to `feature_array`, `num_vec` also needs to be changed, as get_num_vectors needs `num_vec`.
 https://github.com/shogun-toolbox/shogun/blob/a075792c536d67e295927577d12b7bf335565550/src/shogun/features/CombinedFeatures.h#L70-L75

Comment (karlnapf) [reply src/shogun/features/CombinedFeatures.cpp]: nice!
I think in fact num_vec should never be stored ... only every queried ...but this is ok as well. thx

**Comment by karlnapf** [comment]
LGTM

File: examples/meta/src/binary/mkl_binclass.sg.in [hunk 2]
```diff
@@ -36,7 +36,7 @@ mkl.train()
 Features feats_pred = create_features("CombinedFeatures")
```
Comment (karlnapf) [inline examples/meta/src/binary/mkl_binclass.sg.in]: minor: could you change the filename to be consistent with the style we use. In this  case simply "multiple_kernel_learning.sg" (folder is already called binarry)

File: examples/meta/src/binary/mkl_binclass.sg.in [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+#![create_features]
+Features features_train = create_features(f_feats_train)
+Features features_test = create_features(f_feats_test)
+Labels labels_train = create_labels(f_labels_train)
+#![create_features]
+
+#![create combined train features]
+Kernel tkernel = create_kernel("PolyKernel", cache_size=10, degree=3)
+tkernel.init(features_train, features_train)
```
Comment (karlnapf) [inline examples/meta/src/binary/mkl_binclass.sg.in]: something is messed up here

File: examples/meta/src/binary/mkl_binclass.sg.in (hunk 2 above)
Comment (LiuYuHui) [reply examples/meta/src/binary/mkl_binclass.sg.in]: it seems like we already have a meta example called "multiple_kernel_learning.sg".

Comment (karlnapf) [reply examples/meta/src/binary/mkl_binclass.sg.in]: in the "binary" folder? If it is in another one, that is fine as the meta examples are named as foldername-example_name in cmake

Comment (karlnapf) [reply examples/meta/src/binary/mkl_binclass.sg.in]: ah no it is in the same folder...
Could you then check whether they are doing the same thing (maybe we just forgot to delete it). If so, no need to add another one. If there is something that is missing from the old one, feel free to merge the two (if it makes sense).

Comment (gf712) [reply examples/meta/src/binary/mkl_binclass.sg.in]: @LiuYuHui  have you checked if there is anything that needs to be added? If this doesn't add anything can just delete the legacy example and no need to change the meta example

File: examples/meta/src/binary/mkl_binclass.sg.in (hunk 3 above)
Comment (gf712) [reply examples/meta/src/binary/mkl_binclass.sg.in]: in what way?

File: examples/meta/src/binary/mkl_binclass.sg.in (hunk 2 above)
Comment (LiuYuHui) [reply examples/meta/src/binary/mkl_binclass.sg.in]: not completely same, but very similar

Comment (gf712) [reply examples/meta/src/binary/mkl_binclass.sg.in]: if nothing new is tested then just keep the old meta example and delete the legacy example as well as the one you wrote

**Pull request merged**